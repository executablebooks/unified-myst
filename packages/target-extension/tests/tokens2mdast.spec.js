import { fromMarkdown } from 'mdast-util-from-markdown'

import { mystTargetMmarkExt, mystTargetMdastExt } from '../src/index.js'

const fixtures = [
    '(abc)=',
    ' (abc)=',
    '(abc)= ',
    '(a bc)=',
    String.raw`(ab\\c)=`,
]

describe('Convert source to MDAST', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystTargetMmarkExt],
            mdastExtensions: [mystTargetMdastExt],
        }
        const mdast = fromMarkdown(content, options)
        expect(mdast).toMatchSnapshot()
    })
})
