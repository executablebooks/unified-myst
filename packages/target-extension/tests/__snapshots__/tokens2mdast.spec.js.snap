// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to MDAST " (abc)=" 1`] = `
Object {
  "children": Array [
    Object {
      "label": "abc",
      "position": Object {
        "end": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
        "start": Object {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "mystTarget",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "(a bc)=" 1`] = `
Object {
  "children": Array [
    Object {
      "label": "a bc",
      "position": Object {
        "end": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystTarget",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "(ab\\\\\\\\c)=" 1`] = `
Object {
  "children": Array [
    Object {
      "label": "ab\\\\c",
      "position": Object {
        "end": Object {
          "column": 9,
          "line": 1,
          "offset": 8,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystTarget",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "(abc)= " 1`] = `
Object {
  "children": Array [
    Object {
      "label": "abc",
      "position": Object {
        "end": Object {
          "column": 7,
          "line": 1,
          "offset": 6,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystTarget",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Convert source to MDAST "(abc)=" 1`] = `
Object {
  "children": Array [
    Object {
      "label": "abc",
      "position": Object {
        "end": Object {
          "column": 7,
          "line": 1,
          "offset": 6,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "mystTarget",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 7,
      "line": 1,
      "offset": 6,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
